daemon/cmd/daemon_main.go

```diff
@@ -1386,8 +1386,13 @@ func initEnv(vp *viper.Viper) {
 		option.Config.EncryptInterface = append(option.Config.EncryptInterface, link)
 	}
 
-	if option.Config.TunnelingEnabled() && option.Config.EnableAutoDirectRouting {
-		log.Fatalf("%s cannot be used with tunneling. Packets must be routed through the tunnel device.", option.EnableAutoDirectRoutingName)
+	if option.Config.EnableAutoDirectRouting {
+		if option.Config.TunnelingEnabled() {
+			log.Fatalf("%s cannot be used with tunneling. Packets must be routed through the tunnel device.", option.EnableAutoDirectRoutingName)
+		}
+		if option.Config.IPAM != ipamOption.IPAMKubernetes {
```

Comment:
Are you sure that `autoDirectNodeRoutes` is not compatible with `ClusterPool` IPAM mode as well? IMO it should be, also based on comment in https://github.com/cilium/cilium/commit/2ca717e34477 :

> In ClusterPool mode, the CiliumNode PodCIDR field is populated by
   cilium-operator.

I am also not sure about the `MultiPool` mode:

> In other PodCIDR based IPAM modes (such as ClusterPool or MultiPool), the operator will set the PodCIDRs of the CiliumNode

(https://github.com/cilium/cilium/blob/main/pkg/nodediscovery/nodediscovery.go#L412-L416)

As long as `PodCIDRs` are set on the `CiliumNode` by any means, `autoDirectNodeRoutes` should work IMO.

The documentation for multi-pool (`Documentation/network/concepts/ipam/multi-pool.rst`) even explicitly suggests using `auto-direct-node-routes`.